### PR TITLE
Use forks for PHP 8.2 compatibility until respective PRs merged

### PR DESCRIPTION
**Pull request by pronskiy**
PRs:
- https://github.com/sculpin/sculpin/pull/480
- ~~https://github.com/dflydev/dflydev-canal/pull/1~~
- https://github.com/sculpin/sculpin/pull/482

### CODE REVIEW

**Comment by saundefined** [comment]
LGTM, but shouldn't we also update PHP version in `deploy-site.yml`, `netlify.yml`, and `README.md` to 8.1+?

**Comment by pronskiy** [comment]
For some reason, shivammathur/setup-php installs PHP 8.0.30 even tho it's clearly 8.2 in the yml. I'll debug.

**Comment by saundefined** [comment]
Checked it now on a test repository that used 8.2 in the master branch and 8.1 in the PR
At startup on the `pull_request_target` event, 8.2 was using, after merge - 8.1 🤔 

So.. I guess it runs `main`'s workflow using 8.0

File: composer.json [hunk 1]
```diff
@@ -10,27 +10,35 @@
...
   "require": {
-    "php": "^8.0",
-    "sculpin/sculpin": "^3.1",
-    "twig/twig": "2.15.3",
+    "php": "^8.2",
     "ext-gd": "*",
-    "ext-dom": "*"
+    "ext-dom": "*",
+    "twig/twig": "2.15.3",
+    "sculpin/sculpin": "dev-main#467ffd4d7d0e3b28c8b8234d7f29bb3c8caf6c44",
```
Comment (saundefined) [inline composer.json:28]:
```suggestion
    "sculpin/sculpin": "3.3.0-alpha",
```

Sculpin `3.3.0-alpha` includes sculpin/sculpin@467ffd4, and twig 3 support
(perhaps we can wait to a stable release)

File: composer.json [hunk 2]
```diff
@@ -10,27 +10,35 @@
...
+    "dflydev/canal": "dev-master"
```
Comment (saundefined) [inline composer.json:29]:
```suggestion
```

3.0.0-alpha also contain sculpin/sculpin@fc822d9, so I guess we can drop `canal` either

File: composer.json (hunk 1 above)
Comment (saundefined) [reply composer.json:28]: Go ahead 👍

**Pull request merged**